### ISSUE TITLE
Call RC phone number, `softphone.on('track'` event triggered too early

### ISSUE THREAD

**Issue by tylerlong** [tylerlong/ringcentral-softphone-js#11]
Call RC phone number, `softphone.on('track'` event triggered as soon as the phone is ringing.

Call other numbers, `softphone.on('track'` event not triggered until user picks up the phone call.

Call Zoom phone number cannot reproduce this issue. So It seems to be specific to RingCentral.

Based on the facts I think the softphone SDK is fine.
The root cause could be RingCentral server side. It starts sending phone client (silent) audio track as soon as the phone rings.
The correct behavior should be: don't start the audio track until the phone is answered.

**Comment by tylerlong** [comment]
And there is a confirmed issue about **platform telephony events**: 
> It is a known issue that calls from RC phone number to any other RC phone number will get “Answered”  immediately even the callee has not yet answered the call.

I think the WebRTC issue causes the platform telephony events issue, or at least they are related.

**Comment by tylerlong** [comment]
Could it be a feature named "Early Media":  

- https://tools.ietf.org/html/draft-sen-sip-earlymedia-00
- https://webrtcstandards.info/webrtc-early-media/

But this is irrelevant to the question: How do we know when user answered the call?  This is a very important event and lots of business logic depends on it.

For example: play an announcement after user answers the call. If we don't know when user answers the call, the announcement will be played at wrong time (either too early or too late).

**Update**:

Ref:

![image](https://user-images.githubusercontent.com/733544/93639011-2e51e980-f9ad-11ea-85bd-1e4720b710bc.png)

RingCentral `183 Session Progress` message and `200 OK` message have identical SDP. **So for RingCentral, early media and late media are the same media**.

**Comment by tylerlong** [comment]
RingCentral SIP/WebRTC gateway behaves differently than others. When you make a phone call to a RingCentral number, SIP/WebRTC gateway "answers" the call immediately, and starts to play audio "please hold when I try to connect you...". 

As the caller program/app, "please hold when I try to connect you..." is from the callee, but it is actually from the SIP/WebRTC gateway. The caller program doesn't know when the callee answers the call because there is no event.

For other non-RingCentral  SIP/WebRTC gateways, they won't start sending any audio to caller until callee answers the call.  So the audio track event means callee answers the call.

**Comment by tylerlong** [comment]
From pure SIP perspective, `183 Session Progress` means early media while `200 OK` means late media.  As I tested, RingCentral SIP server sends them out almost at the same time. The client receives `200 OK` 50ms later than `183 Session Progress`. and 50ms is too short to be noticeable.

RingCentral SIP server should issue `183 Session Progress` for early media, and issue `200 OK` only if user answers the call. But current issue is RingCentral SIP server issues `200 OK` without waiting for user to answer the phone call.

![image](https://user-images.githubusercontent.com/733544/93640258-2004cd00-f9af-11ea-99bb-9f4ac9280dc8.png)

**Comment by tylerlong** [comment]
Another issue report: https://community.ringcentral.com/questions/102088/sip-integration-knowing-when-a-call-is-answered.html
